### PR TITLE
net: lib: nrf_cloud: Fix sensor data ack event

### PR DESCRIPTION
**Pull request by jayteemo**
The event NRF_CLOUD_EVT_SENSOR_DATA_ACK was previously not
implemented.  It has been implemented using a tag field
in the `nrf_cloud_sensor_data` struct.
This also cleans up handling of MQTT message ids in the
nrf_cloud library

### CODE REVIEW

File: subsys/net/lib/nrf_cloud/src/nrf_cloud.c [hunk 1]
```diff
@@ -629,15 +625,13 @@ static void api_event_handler(const struct nrf_cloud_evt *nrf_cloud_evt)
...
-		break;
-	case NRF_CLOUD_EVT_SENSOR_ATTACHED:
-		LOG_DBG("NRF_CLOUD_EVT_SENSOR_ATTACHED");
-
 		break;
 	case NRF_CLOUD_EVT_SENSOR_DATA_ACK:
 		LOG_DBG("NRF_CLOUD_EVT_SENSOR_DATA_ACK");
 
 		evt.type = CLOUD_EVT_DATA_SENT;
+		evt.data.msg.buf = (char *)nrf_cloud_evt->data.ptr;
```
[the remark this answers is not in the document]
Comment (plskeggs) [reply subsys/net/lib/nrf_cloud/src/nrf_cloud.c:633]: Agreed

**Review by plskeggs — APPROVED** [review]

File: include/net/nrf_cloud.h [hunk 2]
```diff
@@ -375,7 +376,7 @@ int nrf_cloud_sensor_data_send(const struct nrf_cloud_sensor_data *param);
  * @brief Update the device shadow with sensor data.
  *
  * @param[in] param Sensor data; the data pointed to by param->data.ptr
- *                  must be a cJSON object (cJSON *).
+ *                  must be a valid JSON string.
```
Comment (simensrostad) [inline include/net/nrf_cloud.h:379]: Is this change intended to be in https://github.com/nrfconnect/sdk-nrf/pull/5280/commits/851fbf9a22736c3a3e5ebbb7ef671b540c01dfd3?

**Review by simensrostad — APPROVED** [review]

File: subsys/net/lib/nrf_cloud/src/nrf_cloud.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (jayteemo) [reply subsys/net/lib/nrf_cloud/src/nrf_cloud.c]: removed `CLOUD_EVT_DATA_SENT` from nrf_cloud.c.

**Review by jtguggedal — APPROVED** [review]

**Comment by peknis** [comment]
Always add a tech write to a review when you touch the docs. For the nRF cloud either Mel Weed or me, or both.
